environs/bootstrap/bootstrap.go

```diff
@@ -659,15 +659,33 @@ func isCompatibleVersion(v1, v2 version.Number) bool {
 // for tools syncing, and adds an image metadata source after verifying
 // the contents.
 func setPrivateMetadataSources(metadataDir string) ([]*imagemetadata.ImageMetadata, error) {
-	logger.Infof("Setting default tools and image metadata sources: %s", metadataDir)
-	tools.DefaultBaseURL = metadataDir
+	if _, err := os.Stat(metadataDir); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, errors.Annotate(err, "cannot access metadata directory")
+		}
+		logger.Infof("metadata source: %s, not found", metadataDir)
+		return nil, nil
+	}
+
+	toolsMetadataDir := filepath.Join(metadataDir, storage.BaseToolsPath)
+	if _, err := os.Stat(toolsMetadataDir); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, errors.Annotate(err, "cannot access tools metadata")
+		}
+		logger.Infof("default tools metadata source: %s", tools.DefaultBaseURL)
+	} else {
+		logger.Infof("setting default tools metadata source: %s", toolsMetadataDir)
+		tools.DefaultBaseURL = metadataDir
```

Comment:
we only want to do this if metadataDir ends with /tools
so the logic for this block should be:

toolsMetadataDir = metadataDir
if not toolsMetadataDir endswith /tools then
  toolsMetadataDir += /tools
if toolsMetadataDir exists then
  tools.DefaultBaseURL = toolsMetadataDir